Amplify/Categories/API/Operation/RetryableGraphQLOperation.swift

```diff
@@ -178,9 +195,35 @@ public final class RetryableGraphQLSubscriptionOperation<Payload: Decodable>: Op
     }
 
     public func shouldRetry(error: APIError?) -> Bool {
-        return attempts < maxRetries
+//        return attempts < maxRetries
+        
+        guard case let .operationError(errorDescription, recoverySuggestion, underlyingError) = error else {
+            return false
+        }
+        
+        if let authError = underlyingError as? AuthError {
+            switch authError {
+            case .signedOut, .notAuthorized:
+                return attempts < maxRetries
+            default:
+                return false
+            }
+        }
+        
+        // TODO: - How to distinguish errors?
+        // TODO: - Handle other errors
+        if error.debugDescription.contains("Filters combination exceed maximum limit 10 for subscription.") &&
+            filterLimitRetried == false {
+            
+            // Just to be sure that endless retry won't happen
+            filterLimitRetried = true
+            maxRetries += 1
```

Comment:
is `maxRetries` updated here to account for the `attempt` being incremented for the RTF retry case? So the overall `attempt` to `maxRetries` counts will continue to represent the number of auth types to attempt? (maxRetries is set to the number of auth types to attempt in multi-auth rules scenarios).